sycl/source/detail/program_manager/program_manager.hpp

```diff
@@ -232,7 +235,11 @@ class ProgramManager {
                              const std::string &KernelName);
 
   // The function returns the unique SYCL kernel identifier associated with a
-  // kernel name.
+  // kernel name or nullopt if there is no such ID.
+  std::optional<kernel_id> tryGetSYCLKernelID(const std::string &KernelName);
+
+  // The function returns the unique SYCL kernel identifier associated with a
+  // kernel name or throws a sycl exception if there is no such ID.
   kernel_id getSYCLKernelID(const std::string &KernelName);
```

Comment:
I'm not 100% but can these getters be `const`?

Reply:
Because they take the mutex, I don't think they can be `const`.